### PR TITLE
arrays-programs_involving_multidimensional_data-easy-q9

### PR DESCRIPTION
**Pull request by KaiKhademi**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”
---
## Website
- [ ] Is the question text on the Gamification document up to date? 
    > Make sure to include the sample output if there is one for the question

### CODE REVIEW

File: src/arrays/programs_involving_multidimensional_data/easy/q9/Sum.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package arrays.programs_involving_multidimensional_data.easy.q9;
+
+/*
+Write a Java program that generates a 2D array of random integers of size 5 by 5, with the range of values from 100-1000
```
Comment (SamamthaH) [inline src/arrays/programs_involving_multidimensional_data/easy/q9/Sum.java:4]: Change to:
"Write a Java program that generates a 5 by 5 2D array of random integers ranging from 100-1000 (inclusive on both ends), and print the 2D array each row on a new line. Divide all the values by 5, then sum all the values in each row and store those sums into a 1D array which is printed to the console."

File: src/arrays/programs_involving_multidimensional_data/easy/q9/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        }
+        return ans;
+    }
+
+    @Test
+    public void printsCorrectOutput() throws InvalidClauseException {
+        int[][] arr = new int[n][n];
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                arr[i][j] = Integer.parseInt(getItemByName(i + " " + j));
```
Comment (SamamthaH) [inline src/arrays/programs_involving_multidimensional_data/easy/q9/MainTest.java:53]: Consider moving 
```Java
arr[i][j] /= div;
ans[i] += arr[i][j];
```
into this loop and creating the int[] ans up with arr

File: src/arrays/programs_involving_multidimensional_data/easy/q9/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            }
+        }
+        int[] ans = answerFor(arr);
+        int[] out = new int[n];
+        for (int i = 0; i < n; i++) {
+            out[i] = Integer.parseInt(getItemByName(i + ""));
+        }
+        CustomAssertions._assertArrayEquals(ans, out, "Your program does not correctly print the array including sum of each rows after division.");
```
Comment (SamamthaH) [inline src/arrays/programs_involving_multidimensional_data/easy/q9/MainTest.java:61]:
```suggestion
        CustomAssertions._assertArrayEquals(ans, out, "Your program does not correctly print the sum of each modified row.");
```

**Review by SamamthaH — CHANGES_REQUESTED** [review]

**Review by Hedgemon4 — COMMENTED** [review]
See discord messages

File: src/arrays/programs_involving_multidimensional_data/easy/q9/Sum.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package arrays.programs_involving_multidimensional_data.easy.q9;
+
+/*
+Write a Java program that generates a 20 by 20 2D array of random integers ranging from 100-1000 (inclusive on both ends),
+and print the 2D array for each row on a new line. Divide all the values by 5, then sum all the values in each row and
+store those sums into a 1D array printed to the console.
+ */
+
```
Comment (Hedgemon4) [inline src/arrays/programs_involving_multidimensional_data/easy/q9/Sum.java:8]: We should definitely have a sample output for this question

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/arrays/programs_involving_multidimensional_data/easy/q9/Sum.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package arrays.programs_involving_multidimensional_data.easy.q9;
+
+/*
+Write a Java program that generates a 20 by 20 2D array of random integers ranging from 100-1000 (inclusive on both ends),
+and print the 2D array for each row on a new line. Divide all the values by 5, then sum all the values in each row and
```
Comment (Hedgemon4) [inline src/arrays/programs_involving_multidimensional_data/easy/q9/Sum.java:5]:
```suggestion
and print the 2D array with each row a seperate line. Divide all the values by 5, then sum all the values in each row and
```

File: src/arrays/programs_involving_multidimensional_data/easy/q9/Sum.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+store those sums into a 1D array printed to the console.
```
Comment (Hedgemon4) [inline src/arrays/programs_involving_multidimensional_data/easy/q9/Sum.java:6]:
```suggestion
store those sums into a 1D array. Print the sums array to the console.
```

**Review by Hedgemon4 — DISMISSED** [review]

File: src/arrays/programs_involving_multidimensional_data/easy/q9/Sum.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
+package arrays.programs_involving_multidimensional_data.easy.q9;
+
+/*
+Write a Java program that generates a 20 by 20 2D array of random integers ranging from 100-1000 (inclusive on both ends),
+and print the 2D array with each row a seperate line. Divide all the values by 5, then sum all the values in each row and
+store those sums into a 1D array. Print the sums array to the console.
```
Comment (Hedgemon4) [inline src/arrays/programs_involving_multidimensional_data/easy/q9/Sum.java:6]:
```suggestion
store those sums into a 1D array. Print the array of sums to the console.
```

**Pull request merged**